src/transformers/trainer.py

```diff
@@ -2226,16 +2226,33 @@ def _load_best_model(self):
                         state_dict["_smp_is_partial"] = False
                         load_result = model.load_state_dict(state_dict, strict=True)
                 else:
-                    # We load the model state dict on the CPU to avoid an OOM error.
-                    if self.args.save_safetensors and os.path.isfile(best_safe_model_path):
-                        state_dict = safetensors.torch.load_file(best_safe_model_path, device="cpu")
+                    if hasattr(model, "base_model") and getattr(model.base_model, "is_8bit_serializable", False):
```

Comment:
What if the model itself has `is_8bit_serializable`? `base_model` is for models with head but sometimes it's the base model that is passed to the Trainer.

Reply:
As [@younesbelkada](https://github.com/younesbelkada) mentiones in  https://github.com/huggingface/transformers/pull/23443#pullrequestreview-1434190699 
> You can't do pure 8bit training using transformers, the only possible scenario where you can train base 8 bit models is when using PEFT & LoRA. 

So if the model itself has is_8bit_serializable, there is an warnning in line 2244
``` python
logger.warning("Could not loading a quantized checkpoint.")
```